presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -314,18 +315,50 @@ public FileStatus getFileStatus(Path path)
         }
 
         return new FileStatus(
-                getObjectSize(metadata),
+                getObjectSize(metadata, path),
                 false,
                 1,
                 BLOCK_SIZE.toBytes(),
                 lastModifiedTime(metadata),
                 qualifiedPath(path));
     }
 
-    private static long getObjectSize(ObjectMetadata metadata)
+    private long getObjectSize(ObjectMetadata metadata, Path path) throws IOException
     {
         String length = metadata.getUserMetadata().get(UNENCRYPTED_CONTENT_LENGTH);
-        return (length != null) ? Long.parseLong(length) : metadata.getContentLength();
+
+        if (length != null) {
+            return Long.parseLong(length);
+        }
+
+        // check if file has KMS encryption
```

Comment:
most/all of the comments in this method are unnecessary as the code is pretty self explanatory.